src/Microsoft.VisualStudio.Services.Agent/TraceManager.cs

```diff
@@ -71,6 +71,10 @@ private void Dispose(bool disposing)
         private Tracing CreateTraceSource(string name)
         {
             SourceSwitch sourceSwitch = Switch;
+
+            // Flush it to avoid loosing the log if the process is killed
+            Trace.AutoFlush = true;
```

Comment:
Right now if you start agent listener and kill it using task explorer, all the logs are lost. Thats why I made this change. But this is a bad idea I know, any suggestion are welcome,